### PR TITLE
Math Magicians: Events

### PR DESCRIPTION
**Pull request by hernandanielzamora**
In this PR, changes were made to the project: 
- Created events for the calculator component (calculations)
- Used big.js library for the logic part (using calculate.js and operate.js files)

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,38 +1,53 @@
...
+            <button type="button" className="calc-btn" onClick={handleClick}>3</button>
+            <button type="button" className="calc-btn calculation-btn" onClick={handleClick}>+</button>
+          </div>
+          <div>
+            <button type="button" className="calc-btn zero-btn" onClick={handleClick}>0</button>
+            <button type="button" className="calc-btn" onClick={handleClick}>.</button>
+            <button type="button" className="calc-btn calculation-btn" onClick={handleClick}>=</button>
+          </div>
         </div>
       </div>
```
Comment (julie-ify) [inline src/components/Calculator.js]: - [ ] You were required to create a child component in a previous project. Now in this project, you are required to pass props to the child component. I cannot find any child components here. Kindly create a child component and use it as a child of this calculator component in any way you deem fit, but the state should remain in this Calculator component.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @hernandanielzamora,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,38 +1,52 @@
...
 
 export default function CalculatorComp() {
+  const [currentState, setState] = useState({ total: 0, next: null, operation: null });
+
+  const handleClick = (e) => {
+    setState((currentState) => calculate(currentState, e.target.textContent));
+    if (e.target.textContent === 'AC') {
+      setState({ total: 0, next: null, operation: null });
+    }
+  };
```
Comment (addod19) [inline src/components/Calculator.js]: - [ ] I can not find where we passed `props` from the parent component to the child component in our application as requested from the previous code reviewer. For more information about `props`, kindly visit this [link](https://react.dev/learn/passing-props-to-a-component)

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @hernandanielzamora, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Correct Git-flow followed :100: 
- [ ] README is descriptive :rocket: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 2 above)
Comment (hernandanielzamora) [reply src/components/Calculator.js]: @addod19 Hello! I'll really appreciate it if you can clear some doubts about this project. 
I think I've built a child component already. This child component is created in another file, named CalculatorComponent.js
That's why you are not going to find the child in this Calculator.js, Can you check the CalculatorComponent.js, please? So you can tell me if this is okay.

File: src/components/Calculator.js (hunk 1 above)
Comment (hernandanielzamora) [reply src/components/Calculator.js]: @julie-ify Hello! I'll really appreciate it if you can clear some doubts about this project. 
I think I've built a child component already. This child component is created in another file, named CalculatorComponent.js
That's why you are not going to find the child in this Calculator.js, Can you check the CalculatorComponent.js, please? So you can tell me if this is okay

File: src/components/Calculator.js (hunk 2 above)
Comment (hernandanielzamora) [reply src/components/Calculator.js]: @addod19 Here are some screenshots of what I'm talking about
event-react-1
react-event-2

File: src/components/Calculator.js (hunk 1 above)
Comment (hernandanielzamora) [reply src/components/Calculator.js]: @julie-ify Here are some screenshots of what I'm talking about
event-react-1
react-event-2

File: src/components/Calculator.js (hunk 2 above)
Comment (hernandanielzamora) [reply src/components/Calculator.js]: @addod19 Or I'm supposed to do it like this??

react-event-3
react-event-4

File: src/components/Calculator.js (hunk 1 above)
Comment (hernandanielzamora) [reply src/components/Calculator.js]: @julie-ify Or I'm supposed to do it like this?
react-event-3
react-event-4

**Review by andersonlebon — APPROVED** [review]
## Hi @hernandanielzamora 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**